### ISSUE TITLE
[Spike] Form controls, Toolbar controls, and atomic controls

### ISSUE THREAD

**Issue by mjhenkes** [cerner/terra-core#3343]
# Feature Request

## Description

Today terra offers form controls, which include both form field versions and 'raw' versions, these controls are used anywhere that the control would visually be expected to be, but not where one would expect forms to be. 

With this Spike we are proposing to formally separate form controls, toolbar controls and atomic (formerly referred to as raw) controls into their own groups of packages, either packaged all together or packaged separately. 

### Form controls

The form controls group should include any control intended to be placed in a form. The controls should only be available wrapped in a form field element that includes labels, help text, errors and any accessibility concerns. The form controls group should also include a generic form field component to be used to wrap custom form controls. This group should also include a form layout to aid consumers in easily creating forms.

### Toolbar controls

The toolbar controls group should include any control intended to be placed in a toolbar. These components are expected to have unique styling with a focus on being compact as opposed to the more vertical forms controls. Controls like radio buttons either will have a unique design or not be offered. toolbar controls will also need to have accessibility concerns baked in and offer help and error text when appropriate. Part of this spike should be determining if there should be toolbar layouts.

### Atomic controls

Atomic controls are intended to be used in the creation of other higher order components. They will be the basic control with styling applied. In most cases atomic controls will offer nothing beyond the basic html control's api and any accessibility concerns will have to be handled by the consumer. Atomic controls are an advanced usage control.
